### PR TITLE
[migrator] When renaming a function decl, we should use underscore to represent empty external argument label. rdar://34569243

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@nathawes could you review this for swift-4.0-branch?

File: lib/Migrator/APIDiffMigratorPass.cpp [hunk 1]
```diff
@@ -695,9 +695,10 @@ struct APIDiffMigratorPass : public ASTMigratorPass, public SourceEntityWalker {
           if (ArgLoc.isInvalid())
             Editor.insertBefore(PD->getNameLoc(),
                                 (llvm::Twine(NewArg) + " ").str());
```
Comment (nathawes) [inline lib/Migrator/APIDiffMigratorPass.cpp]: Should there be a ternary checking NewArg is empty here too?

Comment (nkcsgexi) [reply lib/Migrator/APIDiffMigratorPass.cpp]: Good catch, we've not seen this kind of change though. Will add here too.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Review by nathawes — APPROVED** [review]
Looks good Xi!

**Pull request merged**